service/history/historyEngine.go

```diff
@@ -345,38 +345,51 @@ func (e *historyEngineImpl) registerDomainFailoverCallback() {
 func (e *historyEngineImpl) createMutableState(
 	clusterMetadata cluster.Metadata,
 	domainEntry *cache.DomainCacheEntry,
-) mutableState {
+	runID string,
+) (mutableState, error) {
 
 	domainName := domainEntry.GetInfo().Name
-	if e.config.EnableEventsV2(domainName) && e.config.EnableNDC(domainName) {
-		return newMutableStateBuilderWithVersionHistories(
+	enableEventsV2 := e.config.EnableEventsV2(domainName)
+	enableNDC := e.config.EnableNDC(domainName)
+
+	var newMutableState mutableState
+	if enableEventsV2 && enableNDC {
```

Comment:
So if event v2 is disable, it won't create mutable state with version histories?

Reply:
yes, how does it work without events v2?